src/helpers/validationRules.ts

```diff
@@ -0,0 +1,53 @@
+export interface ValidationRule {
+  isValid: (val: string) => boolean;
+  message: string;
+  visible?: boolean;
+}
```

Comment:
naaajs! 👏 

[just a note] at some point it may be useful to allow other types of `val` - in that case you can easily extend it with generics e.g.

```suggestion
export interface ValidationRule<Val = string> {
  isValid: (val: Val) => boolean;
  message: string;
  visible?: boolean;
}
```

note that because the default is set to `string` you don't need any changes in current definitions :)

Reply:
Yeah, this will make it more universal, thanks for the tip